src/scripting/game_lua_kernel.cpp

```diff
@@ -4024,6 +4025,36 @@ const game_events::queued_event & game_lua_kernel::get_event_info() {
 	return *queued_events_.top();
 }
 
+/**
+	* @brief takes a formated string and put the variables
+	* @example wesnoth.format("I have $gold, and I need $goldNeed", {['gold'] = '100', ['goldNeed'] = '50'})
+	*/
+static int intf_format(lua_State *L)
+{
+	// check if both arguments are set
+	if (lua_isnoneornil(L, 1) && lua_isnoneornil(L, 2)) {
```

Comment:
It's not necessary to check that the arguments exist, because if they don't then `lua_istring` and `lua_istable` will return false.